doc/examples/segmentation/plot_fast_draw_segmentation.py

```diff
@@ -0,0 +1,198 @@
+        # generate labels as foreground, 1
+        labels[yy-brush : yy+brush+1, xx-brush : xx+brush+1] = 1
+        # refresh plot
+        pltimg.set_data(rgb_image)
+        plt.draw()
+    elif event.button == 3:
+        # if right click is detected during the mouse mouvement
+        # get x and y coordinates
+        yy = int(event.ydata)
+        xx = int(event.xdata)
+        # draw green pixels on the rgb image
+        rgb_image[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = (0,255,0)
+        # generate labels as background, 2
+        labels[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = 2
+        # refresh plot
+        pltimg.set_data(rgb_image)
+        plt.draw()
+
+# one click callback function
+def onClick(event):
+    """This function is called when mouse buttons are clicked
+    """
+    # get x and y coordinates
+    yy = int(event.ydata)
+    xx = int(event.xdata)
+    if event.button == 1:
+        # draw red pixels on the rgb image
+        rgb_image[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = (255,0,0)
+        # generate labels as foreground, 1
+        labels[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = 1
+    elif event.button == 3:
+        # draw green pixels on the rgb image
+        rgb_image[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = (0,255,0)
+        # generate labels as background, 2
+        labels[yy-brush:yy+brush+1,xx-brush:xx+brush+1] = 2
+    # refresh plot
+    pltimg.set_data(rgb_image)
+    plt.draw()
+    
+# keyboard callback events
+def onKeypress(event):
+    """This function is called when keyboard buttons are pressed
+    """
+    global target_label
+    if event.key == ' ':
+        # if space bar is pressed, FastDRaW  and RW segmentations are performed
+        
+        # FastDRaW
+        t = time.clock()
+        # --- perform the FastDRaW segmentation
+        segm = fastdraw.update(labels, target_label=target_label, k=0.5)
+        # --- display the contour in red over the original image
+        contour = (dilation(segm, disk(1)) - segm).astype(np.bool)
+        result = original_image.copy()
+        result[contour,:] = (255,0,0)
+        # --- show the image
+        ax2.imshow(result)
+        ax2.set_title("FastDRaW time "+str(time.clock()-t)+" s")
+        plt.draw()
+        
+        # RW
+        t = time.clock()
+        # perform the RW segmentation
+        segm = random_walker(image,labels,beta=beta)
+        # select which label to display
+        # since this is a binary case segmentation (foreground/background),
+        # segm == 1 is similar to segm == 2
+        segm = segm == 1 
+        # --- display the contour in red over the original image
+        contour = (dilation(segm, disk(1)) - segm).astype(np.bool)
+        result = original_image.copy()
+        result[contour,:] = (255,0,0)
+        # --- show the image
+        ax3.imshow(result)
+        ax3.set_title("RW time "+str(time.clock()-t)+" s")
+        plt.draw()
+    
+
+"""
+Main
```

Comment:
What are "callback functions" and "main"? In plaintext parts, try to address novice coders, who might not have heard these terms before. e.g. "First we write functions to define what needs to happen when the user clicks on the image", "Now we write the main analysis script that makes use of those functions."